re_data/notifications/email.py

```diff
@@ -59,6 +59,7 @@ def send_mime_email(
         server = smtplib.SMTP_SSL(smtp_host, smtp_port)
     else:
         server = smtplib.SMTP(smtp_host, smtp_port)
+        server.starttls()
```

Comment:
I think we can have `smtp_starttls` as part of our email configuration. From the documentation of the method, it's possible the server doesn't support it.
https://github.com/python/cpython/blob/main/Lib/smtplib.py#L770-L772

Reply:
@dejii , not really sure what you propose here. If the mail server really does not support tls, you should simply use ssl? In mine case and I think in many cases you should use TLS when using the default mail port.

```text
Port 587:
This is the default mail submission port. When users submit an email to be routed by a proper mail server, this is the one that will provide best results.

Port 587, coupled with TLS encryption, ensures that email is submitted securely and following the guidelines set out by the IETF.

All Mailgun customers should consider using port 587 as their default SMTP port unless you're explicitly blocked by your upstream network or hosting provider.
```

See [this](https://www.mailgun.com/blog/email/which-smtp-port-understanding-ports-25-465-587/) for more info.

Reply:
@dejii , added it as a flag to the mail configuration. Left the default value for ssl (`true`) as it was in order to not break any existing setup for users. Designed it that if you set both `use_ssl` and `use_tls` to true that it prefers tls. If both are set to `false`, then you will just use a non-encrypted connection.

I was doubting if I should document this behavior somewhere?

Reply:
Yes, you can help document this [here](https://github.com/re-data/re-data/blob/master/docs/docs/re_data/reference/notifications/configuring_channels.md). I see you have already done that

If both `use_tls` and `use_ssl` are set to true, the current logic would go with `use_ssl`. I believe you meant to use an `elif` statement.

Lastly I think we should separate the getting the SMTP connection from the use of TLS, this way whatever you configure is what gets initialized. So we end up with;
```python
    if use_ssl:
        server = smtplib.SMTP_SSL(smtp_host, smtp_port)
    else:
        server = smtplib.SMTP(smtp_host, smtp_port)
    if use_tls:
        server.starttls()
```

Reply:
Yeah I forgot to use the `elif`. I think your code looks clean, but it do think that you should prefer tls over ssl.

> TLS is the new protocol for secured encryption on the web maintained by IETF/SSL consisting of two layers within the layer. TLS is an improved version of SSL. It works in the same way as the SSL using encryption got protection. SSL and TLS are often used interchangeably. TLS uses a stronger encryption algorithm and can work on different ports. However, TLS 1.0 does not work with SSL 3.0.

Looked at your sample and it does make sense to me. But you can't call `server.starttls()` on `server = smtplib.SMTP_SSL(smtp_host, smtp_port)`. See [this](https://docs.python.org/3/library/smtplib.html) for more info.

> An [SMTP_SSL](https://docs.python.org/3/library/smtplib.html#smtplib.SMTP_SSL) instance behaves exactly the same as instances of [SMTP](https://docs.python.org/3/library/smtplib.html#smtplib.SMTP). [SMTP_SSL](https://docs.python.org/3/library/smtplib.html#smtplib.SMTP_SSL) should be used for situations where SSL is required from the beginning of the connection and using starttls() is not appropriate.

@dejii, made a new proposal, let me know what you think.

Reply:
You're right on that. Thanks! LGTM.